### PR TITLE
Big Rewrite - Use MARLlib for everything

### PR DESCRIPTION
**Pull request by sbinabdullah**
- [x] Implement basic skeleton to work around 
- [x] Get forward pass working (MOSTLY DONE)
- [ ] Get backward pass (training) working
    - [x] **[IN_PROGRESS]** Seperate residual and actor network grad update logic from that of the belief network. I believe this can be done by modifying how the `compute_gradients()` function of the policy, `optimizer()`, as well as some other smaller subroutines. 
 - [x] Implement the different loss functions for different parts of the policy.
 - [ ] Optional(???) Modify the specifics of the simple adversary environment to match the environment described by Oguntola et al.

### CODE REVIEW

**Comment by sbinabdullah** [comment]
Just updated the tasks for this PR. I think I've figured out how everything is structured w.r.t. MARLLib, Rllib, etc. There's some small things regarding the theory behind the training process proposed by Oguntola et. el that I need clarification on. Apart from that I should be on the  home straight with regards to implementing their work.

**Comment by sbinabdullah** [comment]
Alright, all that's left is trying to get the backward pass working and we have a working implementation of Oguntola et al.s' work to play around with!

**Comment by sbinabdullah** [comment]
I'm going to merge this and leave it as is for now. Maybe I'll have the chance to  finish the rest of this off and continue my work on it some other time :)
